# Ubuntu: https://bugs.launchpad.net/ubuntu/+source/policykit-gnome/+bug/332343
# Upstream: http://bugzilla.gnome.org/show_bug.cgi?id=572878
# Patch: http://bugzilla.gnome.org/attachment.cgi?id=129353&action=view
# Description: When "clicked" is blocked as the user is not authorized
#   it should emit "auth-start" so that the user can authenticate. Makes
#   it work with widgets that don't emit "activate", e.g. buttons.
--- ./polkit-gnome/polkit-gnome-action.c.old	2009-02-06 17:31:46.000000000 +0000
+++ ./polkit-gnome/polkit-gnome-action.c	2009-02-06 17:32:03.000000000 +0000
@@ -2062,6 +2062,8 @@ _button_clicked (GtkButton *button, PolK
         case POLKIT_RESULT_ONLY_VIA_SELF_AUTH_KEEP_ALWAYS:
                 /* g_debug ("blocking clicked"); */
                 g_signal_stop_emission_by_name (button, "clicked");
+                if (action->priv->polkit_action != NULL)
+                        g_signal_emit (action, signals [AUTH_START_SIGNAL], 0);
                 break;
 
         default:
